src/webvr-polyfill.js

```diff
@@ -13,46 +13,81 @@
  * limitations under the License.
  */
 
-var CardboardHMDVRDevice = require('./cardboard-hmd-vr-device.js');
-//var OrientationPositionSensorVRDevice = require('./orientation-position-sensor-vr-device.js');
-var FusionPositionSensorVRDevice = require('./fusion-position-sensor-vr-device.js');
+var CardboardVRDisplay = require('./cardboard-vr-display.js');
 var MouseKeyboardPositionSensorVRDevice = require('./mouse-keyboard-position-sensor-vr-device.js');
 // Uncomment to add positional tracking via webcam.
 //var WebcamPositionSensorVRDevice = require('./webcam-position-sensor-vr-device.js');
+var VRDisplay = require('./base.js').VRDisplay;
 var HMDVRDevice = require('./base.js').HMDVRDevice;
 var PositionSensorVRDevice = require('./base.js').PositionSensorVRDevice;
+var VRDisplayHMDDevice = require('./display-wrappers.js').VRDisplayHMDDevice;
+var VRDisplayPositionSensorDevice = require('./display-wrappers.js').VRDisplayPositionSensorDevice;
 
 function WebVRPolyfill() {
-  this.devices = [];
+  this.displays = [];
+  this.devices = []; // For deprecated objects
+  this.devicesPopulated = false;
+  this.nativeWebVRAvailable = this.isWebVRAvailable();
 
-  if (!this.isWebVRAvailable()) {
+  if (!this.nativeWebVRAvailable) {
```

Comment:
IMO, we should bail `return` early if `this.nativeWebVRAvailable` (would also prevent having to [check `WebVRConfig.ENABLE_DEPRECATED_API` below](#diff-7bfb91b61bb80cff41aba297525c4dc7R35) if you wanted to).